### ISSUE TITLE
Preferences should have unique key names

### ISSUE THREAD

**Issue by deathaxe** [jisaacks/GitGutter#326]
In the latest commits I found a `GitGutterSettings` class, which makes use of _Preferences.sublime-settings_ to look for GitGutter settings like `git_binary`, `show_in_minimap` or `show_status` .

While `git_path` might be safe to be stored in Preferences as it could be used by other plugins, too, the names of the other two settings do not have a unique meaning. The question is: What to show in the minimap or status? Just imagine another plugin using these settings for own purposes storing them in _Preferences.sublime-settings_ like GitGutter does. So the user would always change the beahaviour of both plugins.

By default calling `sublime.load_sttings("GitGutter.sublime-settings")` returns an object which contains the merged content of all _GitGutter.sublime-settings_ files found by Sublime Text. This way the user already can make overides of single settings by creating a ${User}/GitGutter.sublime-settings file.

If you really want to store some settings in the _Preferences.sublime-settings_ their name must have a unique meaning. I would prefere to use `gitgutter_show_in_minimap` and `gitgutter_show_status` in your case. As mensioned `git_binary` is unique, already.

**Comment by r-stein** [comment]
I think it would be better to prefix them with `gitgutter.`, e.g. `gitgutter.show_status`

**Comment by rchl** [comment]
That is true. The package claims that it supports per-project settings and it attempts to support them but does it wrong. `sublime.load_settings('Preferences.sublime-settings')` won't return project settings. To get project settings you need to access `view.settings()`.

Good thing is that adding the prefix to project settings won't break anything as it wasn't working already.

I would suggest using `git_gutter_*` prefix. Seems in line with the name and file names used in the project and is general convention that I've seen used in various plugins (rather than using a dot).

**Comment by r-stein** [comment]
For example mdpopups uses a dot. I like that more, because it is a better distinction betwen the parts.
E.g. if we have a setting `git_gutter_enabled` it could be the `enabled` setting of `git_gutter` or the `gutter_enabled` setting of `git`. However at the end this is not a real problem and I don't care to much about that detail.

**Comment by rchl** [comment]
I don't mind dot either. It's just I've never seen dot being used before but I've seen other convention being used quite a few times.

I just have a slight tingling fear that dot could be problematic in accessing a JSON parsed to object as in those cases dot is often used to access object properties so having dot in a property could be problematic. But then, if it's not a problem in python/sublime implementation then dot sounds nicer actually.

**Comment by deathaxe** [comment]
> E.g. if we have a setting git_gutter_enabled ...

@r-stein: That's why I suggested `gitgutter_` as prefix. The prefix should be named like the plugin. I also saw dots as separators. But mdpopups is the only one using them so far and I also tend to not like them as they somehow break naming convention. But this is only detail. 

In general I think, settings should have a unique meaning. It is not essential to have a unique prefix for each plugin. As an example think of the growing number of  `theme_...` settings, like `theme_accent_purple`. This setting can be used by several Theme packages in the same way as only one of them will be active at a time. I also saw something like `material_theme_accent_purple` which I found not so good as it only works with material theme being active. Hope you have an idea about what I want to say with that example.

@rchl I never would have came to the conclusion that this should enable per project settings. I am not sure if `view.settings()` is the right object for project settings. As the _API documentation_ says `view.settings()` are private for a view. I guess they are stored in the session but not the project as well as `window.settings()`. To really add per project settings, you would need to use the `window.set_project_data(data)` function. They can read by `window.project_data()`. `data` is an object of `Dict` which solves your prefix discussion as you can add an gitgutter object to the project file like this:
```json
{
   "folders":[],
   "git_gutter": {
       "git_binary": "",
       "show_in_minimap": true,
       "show_status": true
   }
}
```

**Comment by r-stein** [comment]
ST2 has now window settings, hence your solution is ST3 only. `view.settings()` also reads the project specific settings (and also syntax specific and the User settings)
https://www.sublimetext.com/docs/3/settings.html

**Comment by deathaxe** [comment]
@r-stein: That's just perfect, exactly what is needed. Did not yet realize these documentation.

**Comment by rchl** [comment]
Just noticed that some parts of the code handle project settings:
```
        return view.settings().get('git_gutter_compare_against', compare)
```
So this code is already using the `git_gutter_*` prefix. Would could handle both `git_gutter_` and `git_gutter.` prefixes at the same time to preserve backwards compatibility. Or just stick with existing convention.

**Comment by deathaxe** [comment]
As there are `git_gutter_*` prefixed settings in use already, I would not start using new prefixes. I think it should be pretty clear that this prefix belongs to GitGutter.

**Comment by deathaxe** [comment]
All project-specific settings use `git_gutter_` prefix since Fixed in 1.6.0-rc.1
